x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -586,8 +590,34 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
                   intendedTimestamp,
                 });
 
+                const matchingBuildingBlockAlerts =
+                  newAlerts?.length > 0 && getMatchingBuildingBlockAlerts != null
+                    ? newAlerts.flatMap(
+                        (newAlert) =>
+                          getMatchingBuildingBlockAlerts(newAlert?._source) as Array<{
+                            _id: string;
+                            _source: Record<string, string | number | null>;
+                          }>
+                      )
+                    : [];
+
+                const augmentedBuildingBlockAlerts =
+                  getMatchingBuildingBlockAlerts != null && newAlerts?.length > 0
+                    ? await augmentAlerts({
+                        alerts: matchingBuildingBlockAlerts,
+                        options,
+                        kibanaVersion: ruleDataClient.kibanaVersion,
+                        currentTimeOverride,
+                        intendedTimestamp,
+                      })
+                    : [];
+
                 const bulkResponse = await ruleDataClientWriter.bulk({
-                  body: [...duplicateAlertUpdates, ...mapAlertsToBulkCreate(augmentedAlerts)],
+                  body: [
+                    ...duplicateAlertUpdates,
+                    ...mapAlertsToBulkCreate(augmentedAlerts),
+                    ...mapAlertsToBulkCreate(augmentedBuildingBlockAlerts),
```

Comment:
Is there any chance a single alert could make it into multiple of the dereferenced arrays?  We'd end up seeing conflicts on the individual docs, assuming they are updated with version checking (presumably second doc update of the same underlying doc would be a conflict of the first).

Reply:
Good question. The building block alerts and `augmentedAlerts` (which in this case would be the sequence alert) won't ever cross-polinate between the two arrays.

We start off with all building block alerts and then narrow that group down to only building block alerts that are part of the newly created, unsuppressed alert. All other suppressed alerts will have their building block alerts "thrown away" (not created).

Here is the function that partitions sequence alerts and building block alerts: https://github.com/elastic/kibana/pull/189725/files#diff-7bc3e0677cda5acc5376a472970bd2a6da57a71df78cdc0c77f1d4ce3182410aR209-R212

And here is the function determining which building block alerts to generate based on the non-suppressed alert that will be created: https://github.com/elastic/kibana/pull/189725/files#diff-6732940b0b8040ee9259fdd3463d4671afc3cf6e934a42eb56d399cfb447ae03R105-R114

Hope this is helpful!